### PR TITLE
Seq

### PR DESCRIPTION
**Pull request by LeonLiuY**

### CODE REVIEW

File: src/test/java/com/worksap/fig/SeqTest.java [hunk 1]
```diff
@@ -117,6 +124,37 @@ public void testContainsAny() {
...
+                .containsSubSeq(Seqs.newSeq('A', 'B', 'C')));
+
+        assertThrows(NullPointerException.class, () -> Seqs.newSeq('A', 'B', 'C').containsSubSeq(null));
+
+        assertEquals(15, Seqs.newSeq('B', 'B', 'C', ' ', 'A', 'B', 'C', 'D', 'A', 'B', ' ', 'A', 'B', 'C', 'D', 'A', 'B', 'C', 'D', 'A', 'B', 'D', 'E')
+                .indexOfSubSeq(Seqs.newSeq('A', 'B', 'C', 'D', 'A', 'B', 'D')));
+        assertEquals(15, Seqs.newSeq('B', 'B', 'C', ' ', 'A', 'B', 'C', 'D', 'A', 'B', ' ', 'A', 'B', 'C', 'D', 'A', 'B', 'C', 'D', 'A', 'B', 'D', 'E')
+                .lastIndexOfSubSeq(Seqs.newSeq('A', 'B', 'C', 'D', 'A', 'B', 'D')));
+
+        assertEquals(1, Seqs.newSeq(3, 1, 2, 1, 2, 1, 2, 3)
```
Comment (jeoygin) [inline src/test/java/com/worksap/fig/SeqTest.java:148]: No case of empty and larger pattern seq. For unit test, we can consider hamcrest that is used by junit 4.

File: src/main/java/com/worksap/fig/lang/Seq.java [hunk 2]
```diff
@@ -627,14 +653,133 @@ default boolean containsAny(Predicate<T> condition) {
...
+        if (size() < seq.size()) {
+            return -1;
+        }
+
+        //Sunday algorithm
+
+        Map<T, Integer> lastIndex = new HashMap<>();
+        seq.forEachReverse(lastIndex::put);
+
+        int size = size(), endI = size - 1, len = seq.size();
```
Comment (jeoygin) [inline src/main/java/com/worksap/fig/lang/Seq.java:763]: size is no used latter. Size and len are easy to be confused. How about thisSize and ptnSize?

File: src/test/java/com/worksap/fig/SeqTest.java (hunk 1 above)
Comment (LeonLiuY) [inline src/test/java/com/worksap/fig/SeqTest.java:148]: I think "empty and larger pattern seq" are tested above.

File: src/main/java/com/worksap/fig/lang/Seq.java (hunk 2 above)
Comment (LeonLiuY) [inline src/main/java/com/worksap/fig/lang/Seq.java:763]: Thanks, I've changed `size` to `thisSize`

**Pull request merged**